File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/C#LabAssignment/sharplab/View/Commands/RunCommand.cs
using sharplab.Controller;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.View.Commands
{
    class RunCommand : Command
    {
        private Ctrl ctr;
        public RunCommand(string name, string description, Ctrl ctr) : base(name, description)
        {
            this.ctr = ctr;
        }

        public override void Execute()
        {
            ctr.AllStep();
        }
        public override string ToString()
        {
            return this.GetName() + "." + this.GetDescription() + "\n";
        }

    }
}

File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/View/Commands/RunCommand.java
package View.Commands;

import Controller.Ctrl;
import Exceptions.MyViewException;

public class RunCommand extends Command
{
    private Ctrl ctr;
    public RunCommand(String key, String desc, Ctrl ctr)
    {
        super(key, desc);
        this.ctr = ctr;
    }
    public void execute() {
        try{
            ctr.allStep();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}
